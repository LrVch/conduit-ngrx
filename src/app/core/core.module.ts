import { NgModule, Optional, SkipSelf } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HTTP_INTERCEPTORS } from '@angular/common/http';
import { HttpTokenInterceptor } from './interceptors/http.token.interceptor';

import {
  ApiService,
  ArticlesService,
  AuthGuard,
  CommentsService,
  JwtService,
  ProfilesService,
  TagsService,
  UserService,
  CustomEventManager,
  ScrollService,
  DomUtilService
} from './services';
import { EventManager } from '@angular/platform-browser';

@NgModule({
  imports: [
    CommonModule
  ],
  providers: [
    { provide: HTTP_INTERCEPTORS, useClass: HttpTokenInterceptor, multi: true },
    ApiService,
    ArticlesService,
    AuthGuard,
    CommentsService,
    JwtService,
    ProfilesService,
    TagsService,
    UserService,
    ScrollService,
    DomUtilService,
    { provide: EventManager, useClass: CustomEventManager }
  ],
  declarations: []
})
export class CoreModule {
  constructor (@Optional() @SkipSelf() parentModule: CoreModule) {
    if (parentModule) {
      throw new Error(
        'CoreModule is already loaded. Import it in the AppModule only');
    }
  }
 }
