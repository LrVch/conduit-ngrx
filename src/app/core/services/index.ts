export * from './api.service';
export * from './articles.service';
export * from './auth-guard.service';
export * from './comments.service';
export * from './profiles.service';
export * from './tags.service';
export * from './user.service';
export * from './custom-event-manager';
export * from './main-loader.service';
export * from './scroll.service';
export * from './dom-util.service';
export * from './local-storage.service';
export * from './animations.service';
