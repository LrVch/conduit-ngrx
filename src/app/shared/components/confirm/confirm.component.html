<div mat-dialog-content>
  <p>{{ question }}</p>
</div>
<div mat-dialog-actions>
  <button mat-button
          (click)="onNoClick()"
          cdkFocusInitial>{{ rejectText }}</button>
  <button mat-button
          [mat-dialog-close]="true">{{ successText }}</button>
</div>