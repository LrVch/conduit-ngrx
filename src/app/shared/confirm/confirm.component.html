<div mat-dialog-content>
  <p>{{ question }}</p>
</div>
<div mat-dialog-actions>
  <button mat-button
          (click)="onNoClick()"
          cdkFocusInitial>No Thanks</button>
  <button mat-button
          [mat-dialog-close]="true">Ok</button>
</div>