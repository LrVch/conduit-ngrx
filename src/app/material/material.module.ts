import { NgModule } from '@angular/core';
import {
  MatButtonModule,
  MatCheckboxModule,
  MatToolbarModule,
  MatCardModule,
  MatChipsModule,
  MatProgressBarModule,
  MatPaginatorModule,
  MatTabsModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule
} from '@angular/material';

const modules = [
  MatButtonModule,
  MatCheckboxModule,
  MatToolbarModule,
  MatCardModule,
  MatChipsModule,
  MatProgressBarModule,
  MatPaginatorModule,
  MatTabsModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule
];

@NgModule({
  imports: [...modules],
  exports: [...modules]
})
export class MaterialModule { }
