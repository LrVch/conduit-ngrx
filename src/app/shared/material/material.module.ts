import { NgModule } from '@angular/core';
import {
  MatButtonModule,
  MatCheckboxModule,
  MatToolbarModule,
  MatCardModule,
  MatChipsModule,
  MatProgressBarModule,
  MatPaginatorModule,
  MatTabsModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule,
  MatDialogModule,
  MatSnackBarModule
} from '@angular/material';

const modules = [
  MatButtonModule,
  MatCheckboxModule,
  MatToolbarModule,
  MatCardModule,
  MatChipsModule,
  MatProgressBarModule,
  MatPaginatorModule,
  MatTabsModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule,
  MatDialogModule,
  MatSnackBarModule
];

@NgModule({
  imports: modules,
  exports: modules
})
export class MaterialModule { }
