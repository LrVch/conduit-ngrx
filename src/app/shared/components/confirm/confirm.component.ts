import { Component, Inject } from '@angular/core';
import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material';

export interface DialogData {
  question: string;
  successText: string;
  rejectText: string;
}

@Component({
  selector: 'app-confirm',
  templateUrl: './confirm.component.html'
})
export class ConfirmComponent {
  question: string;
  successText: string;
  rejectText: string;

  constructor(
    private dialogRef: MatDialogRef<ConfirmComponent>,
    @Inject(MAT_DIALOG_DATA) public data: DialogData
  ) {
    data = data || {} as DialogData;
    this.question = data.question || '';
    this.successText = data.successText || 'Ok';
    this.rejectText = data.rejectText || 'No Thanks';
  }

  onNoClick(): void {
    this.dialogRef.close();
  }
}
