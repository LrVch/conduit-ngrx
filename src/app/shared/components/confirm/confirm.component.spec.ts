import { async, ComponentFixture } from '@angular/core/testing';

import { ConfigureFn, configureTests } from '../../../lib/testing';

import { ConfirmComponent } from './confirm.component';
import { DebugElement } from '@angular/core';
import { By } from '@angular/platform-browser';
import { MaterialModule } from '../../material/material.module';
import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material';

describe('ConfirmComponent', () => {
    let component: ConfirmComponent;
    let fixture: ComponentFixture<ConfirmComponent>;
    let de: DebugElement;
    let el: HTMLElement;
    let dialogRefSpy;
    let data;

    beforeEach(
        async(() => {
            data = {
                question: 'How do you do?'
            };
            const dialogRefStub = {
                close() { }
            };
            dialogRefSpy = jest.spyOn(dialogRefStub, 'close');
            const configure: ConfigureFn = testBed => {
                testBed.configureTestingModule({
                    declarations: [ConfirmComponent],
                    imports: [MaterialModule],
                    providers: [{ provide: MatDialogRef, useValue: dialogRefStub }, { provide: MAT_DIALOG_DATA, useValue: data }]
                });
            };

            configureTests(configure).then(testBed => {
                fixture = testBed.createComponent(ConfirmComponent);
                component = fixture.componentInstance;
                de = fixture.debugElement;
                el = de.nativeElement;
                fixture.detectChanges();
            });
        })
    );

    it('should create', () => {
        expect(component).toBeTruthy();
    });

    it('should has question property is set to received data', () => {
        expect(component.question).toEqual(data.question);
    });

    it('should show question', () => {
        expect(fixture).toMatchSnapshot();

        component.question = 'Are you a web developer?';
        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
    });

    it('should close dialog when (onNoClick) clicked', () => {
        const buttonDe = de.query(By.css('button'));

        buttonDe.triggerEventHandler('click', null);

        expect(dialogRefSpy).toHaveBeenCalled();
        expect(dialogRefSpy).toHaveBeenCalledTimes(1);
    });

});

