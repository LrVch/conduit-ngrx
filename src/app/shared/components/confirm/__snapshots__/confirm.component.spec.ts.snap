// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmComponent should show question 1`] = `
<app-confirm
  data={[Function Object]}
  dialogRef={[Function Object]}
  question={[Function String]}
  rejectText={[Function String]}
  successText={[Function String]}
>
  <div
    class="mat-dialog-content"
    mat-dialog-content=""
  >
    <p>
      How do you do?
    </p>
  </div><div
    class="mat-dialog-actions"
    mat-dialog-actions=""
  >
    <button
      cdkfocusinitial=""
      class="mat-button"
      mat-button=""
    >
      <span
        class="mat-button-wrapper"
      >
        No Thanks
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger="[object HTMLButtonElement]"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
    <button
      aria-label="Close dialog"
      class="mat-button"
      mat-button=""
      ng-reflect-dialog-result="true"
      type="button"
    >
      <span
        class="mat-button-wrapper"
      >
        Ok
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger="[object HTMLButtonElement]"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
  </div>
</app-confirm>
`;

exports[`ConfirmComponent should show question 2`] = `
<app-confirm
  data={[Function Object]}
  dialogRef={[Function Object]}
  question={[Function String]}
  rejectText={[Function String]}
  successText={[Function String]}
>
  <div
    class="mat-dialog-content"
    mat-dialog-content=""
  >
    <p>
      Are you a web developer?
    </p>
  </div><div
    class="mat-dialog-actions"
    mat-dialog-actions=""
  >
    <button
      cdkfocusinitial=""
      class="mat-button"
      mat-button=""
    >
      <span
        class="mat-button-wrapper"
      >
        No Thanks
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger="[object HTMLButtonElement]"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
    <button
      aria-label="Close dialog"
      class="mat-button"
      mat-button=""
      ng-reflect-dialog-result="true"
      type="button"
    >
      <span
        class="mat-button-wrapper"
      >
        Ok
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
        ng-reflect-centered="false"
        ng-reflect-disabled="false"
        ng-reflect-trigger="[object HTMLButtonElement]"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
  </div>
</app-confirm>
`;
